integration-tests/src/main/java/org/apache/druid/testing/utils/MsqQueryWithResults.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.testing.utils;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class MsqQueryWithResults extends AbstractQueryWithResults<String>
```

Comment:
How do we pass custom context parameters?

Reply:
IMO this class is just to provide the query and the results. I didn't consider the case with custom context parameters, however, I think we can add these as parameters in the methods that utilize this class. I will make the necessary updates.

Reply:
For custom context parameters, the task needs to be submitted with those in the `submitMsqTask` method. This POJO is used in `compareResults` method which merely fetches the reports for the submitted task and compares the result with the ones present in the report.